operator/redisfailover/service/client.go

```diff
@@ -212,6 +214,20 @@ func (r *RedisFailoverKubeClient) DestroySentinelResources(rf *redisfailoverv1.R
 	return err
 }
 
+func (r *RedisFailoverKubeClient) DestroyRemainedRedisNetworkPolicy(rf *redisfailoverv1.RedisFailover) error {
+
+	name := GetRedisNetworkPolicyName(rf)
+
+	if _, err := r.K8SService.GetNetworkPolicy(rf.Namespace, name); err != nil {
+		if errors.IsNotFound(err) {
+			return nil
+		}
```

Comment:
I don't see any tests for this code path. 

What happens when `err` is NOT an `IsNotFound` error? I think in that case, this swallows `err` and tries to delete the NetworkPolicy anyway. Should we handle this case? Thoughts on the following?

```suggestion
		if errors.IsNotFound(err) {
			return nil
		} else {
			return err
		}
```